### DISCUSSION TITLE
Dashboard variables containing multiple values

### DISCUSSION THREAD

**Discussion by amandeep-r** [grafana/grafana-infinity-datasource#658] [Q&A] [answered]
I am trying to create a dashboard variable where the value in the key value pair is an object like a dictionary or list that I can call into elsewhere in the dashboard.

The example below works but does not meet my need since I am only creating a key value pair consisting of strings:

This is the configuration for a dashboard variable, `variable_name`. It returns a list of devices with each json consisting of that device's attributes. This variable is is selected via drop down menu on the dashboard. 
![image](https://github.com/yesoreyeram/grafana-infinity-datasource/assets/39491878/a336a4a8-21cf-43be-b52c-76ac30c058e3)

In this case the keys appear as `Item_name | v1 | SN 123456` and when addressed in the dashboard via `'${dashboard_variable.serial_num}'` returns the serial number from the json queried

However, when I attempt to include more information inside `dashboard_variable` I run into an issue:

The query for `dashboard_variable` is modified as follows:
```
parse-json
| extend "farm_description"=strcat("name", ' | ', "version", ' | SN ',"serial_num")
| project "farm_description", "serial_num", "active"
```

And now `'${dashboard_variable.serial_num}'` no longer returns the attribute serial_num, instead it returns the description: `Item_name | v1 | SN 123456`.  `'${dashboard_variable.active}'` also returns the description.

Can someone please point me to some documentation to help me understand what's happening or suggest a fix?

Thank you.

**Comment by yesoreyeram** [comment, marked as answer]
For the dashboard variables, you just need to return single field.
